src/engine/plugins/isochrone.cpp

```diff
@@ -0,0 +1,330 @@
+                                  forward_duration_offset,
+                                  reverse_duration_offset,
+                                  vcoord,
+                                  vcoord);
+        }
+        if (data.reverse_segment_id.enabled)
+        {
+            if (util::coordinate_calculation::haversineDistance(vcoord, startcoord) >
+                MAX_TRAVEL_DISTANCE_METERS)
+                return;
+            auto datacopy = data;
+            datacopy.forward_segment_id.enabled = false;
+            phantoms.emplace_back(datacopy,
+                                  forward_weight,
+                                  reverse_weight,
+                                  forward_weight_offset,
+                                  reverse_weight_offset,
+                                  forward_duration,
+                                  reverse_duration,
+                                  forward_duration_offset,
+                                  reverse_duration_offset,
+                                  ucoord,
+                                  ucoord);
+        }
+    });
+
+    std::vector<std::size_t> sources;
+    std::vector<std::size_t> destinations;
+
+    sources.push_back(0);
+    destinations.resize(phantoms.size() - 1);
+    std::iota(destinations.begin(), destinations.end(), 1);
+
+    TIMER_STOP(PHANTOM_TIMER);
+    util::Log() << "Make phantom nodes " << TIMER_MSEC(PHANTOM_TIMER);
+
+    // Phase 1 - outgoing, upwards dijkstra search, setting d(v) for all v we visit
+    heaps.InitializeOrClearFirstThreadLocalStorage(facade->GetNumberOfNodes());
+    heaps.InitializeOrClearSecondThreadLocalStorage(facade->GetNumberOfNodes());
+
+    auto &query_heap = *(heaps.forward_heap_1);
+    if (phantoms[0].forward_segment_id.enabled)
+    {
+        query_heap.Insert(phantoms[0].forward_segment_id.id,
+                          -phantoms[0].GetForwardWeightPlusOffset(),
+                          phantoms[0].forward_segment_id.id);
+    }
+    if (phantoms[0].reverse_segment_id.enabled)
+    {
+        query_heap.Insert(phantoms[0].reverse_segment_id.id,
+                          -phantoms[0].GetReverseWeightPlusOffset(),
+                          phantoms[0].reverse_segment_id.id);
+    }
+
+    while (query_heap.Size() > 0)
+    {
+        const NodeID node = query_heap.DeleteMin();
+        const EdgeWeight weight = query_heap.GetKey(node);
+
+        for (auto edge : facade->GetAdjacentEdgeRange(node))
+        {
+            const auto &data = facade->GetEdgeData(edge);
+            if (data.forward)
+            {
+                const NodeID to = facade->GetTarget(edge);
+                const EdgeWeight edge_weight = data.weight;
+
+                BOOST_ASSERT_MSG(edge_weight > 0, "edge_weight invalid");
+                const EdgeWeight to_weight = weight + edge_weight;
+
+                // No need to search further than our maximum radius
+                if (weight > range * 10)
```

Comment:
TL;TR: Is it intentional here to mismatch signed/unsigned types and rely on `signed int` to `unsigned int` implicit conversion (eg. to filter nodes with negative weights)?

In my previous https://github.com/Project-OSRM/osrm-backend/pull/3652#discussion_r132565108 I asked about signed/unsigned mismatch of `range`.
This `weight > range * 10` test is where the difference matters:
* `weight` is `signed int`
* if `range` is `unsigned int` (eg. pulled as `const auto range = parameters.range;` earlier)
then result of comparison test is different than, I assume, expected.

For example, if `weight` value is negative, it is implicitly converted to `unsigned int`, according to the usual arithmetic conversions pattern (C++/5/9), and overflow flips the result loosing some nodes.